### PR TITLE
Add unit tests Distributions.Discrete.Hypergeometric

### PR DESCRIPTION
**Pull request by Freymaurer**
**Please list the changes introduced in this PR**

- testList "Distributions.Discrete.Hypergeometric" with 9 tests ✅ 
  - testing `hypergeoCheckParam` for edge case coverage
  - test all member functions of `Distributions.Discrete.Bernoulli` used in `Distributions.Discrete.bernoulli`
  - do all tests on different bernoulli distributions covering: min, max, default and nan.
- add verbose failure check `hypergeoCheckParam_k` + tests.
- update docs 📚 
  - add example for the Sample() function
  - Change wording to be more in line with function parameters. Should improve understandability.

**Additional contest**

⚠️ CDF still has float as input. This is missleading and should be changed in a bigger rework of the Distribution<'a,'b> interface!

**[Required]** please make sure you checked that
 - [x] The project builds without problems on your machine

**[Optional]**
 - [x] Added unit tests regarding the added features

### CODE REVIEW

File: src/FSharp.Stats/Distributions/Discrete.fs [hunk 1]
```diff
@@ -179,31 +193,37 @@ module Discrete =
...
 
-        /// Computes the probability density function at k, i.e. P(K = k)
-        static member PDF  N K n k =
+        // Rename PMF? https://en.wikipedia.org/wiki/Probability_mass_function
+        // > A probability mass function differs from a probability density function (PDF) in that the latter is associated with continuous 
+        // > rather than discrete random variables. A PDF must be integrated over an interval to yield a probability.
+
+        /// Computes the probability density function at k for P(X = k).
+        static member PDF N K n k =
             hypergeoCheckParam N K n
```
Comment (bvenn) [inline src/FSharp.Stats/Distributions/Discrete.fs:222]: Remove this line to not checking parameters twice.

File: src/FSharp.Stats/Distributions/Discrete.fs [hunk 2]
```diff
@@ -179,31 +193,37 @@ module Discrete =
...
                 exp ((SpecialFunctions.Binomial.coeffcientLn K k) + (SpecialFunctions.Binomial.coeffcientLn (N-K) (n-k)) - SpecialFunctions.Binomial.coeffcientLn N n)
-
-        /// Computes the cumulative distribution function at x, i.e. P(X <= x).
-        static member CDF N K n (x:float) =
-            hypergeoCheckParam N K n            
-            if (x < float (max 0 (n + K - N))) then 
+        
+        /// Computes the cumulative distribution function at k for P(X <= k).
+        static member CDF N K n (k: int) =
+            hypergeoCheckParam N K n
```
Comment (bvenn) [inline src/FSharp.Stats/Distributions/Discrete.fs:231]: see above

**Review by bvenn — CHANGES_REQUESTED** [review]

File: src/FSharp.Stats/Distributions/Discrete.fs (hunk 1 above)
Comment (Freymaurer) [reply src/FSharp.Stats/Distributions/Discrete.fs:222]: Do you mean i should remove `hypergeoCheckParam`? Because `hypergeoCheckParam` and `hypergeoCheckParam_k` test different cases. They do not overlap.

Comment (bvenn) [reply src/FSharp.Stats/Distributions/Discrete.fs:222]: Correct, sorry for the confusion!

**Review by bvenn — APPROVED** [review]
